### PR TITLE
MESOS: configure kube-proxy to run in mesos container instead of /kube-proxy

### PR DESCRIPTION
**Pull request by jdef**
- moved cgroup-root detection to minion service
- kube proxy now configured to run in mesos container

xref https://github.com/mesosphere/kubernetes-mesos/issues/462

### CODE REVIEW

**Comment by jdef** [comment]
@sttts PTAL

File: contrib/mesos/pkg/minion/server.go [hunk 1]
```diff
@@ -277,6 +310,7 @@ func (ms *MinionServer) AddMinionFlags(fs *pflag.FlagSet) {
 	// general minion flags
 	fs.BoolVar(&ms.privateMountNS, "private-mountns", ms.privateMountNS, "Enter a private mount NS before spawning procs (linux only). Experimental, not yet compatible with k8s volumes.")
 	fs.StringVar(&ms.pathOverride, "path-override", ms.pathOverride, "Override the PATH in the environment of the sub-processes.")
+	fs.StringVar(&ms.cgroupPrefix, "cgroup-prefix", ms.cgroupPrefix, "The cgroup prefix concatenated with MESOS_DIRECTORY must give the executor cgroup set by Mesos")
```
Comment (sttts) [inline contrib/mesos/pkg/minion/server.go]: (Default: /mesos)

File: contrib/mesos/pkg/minion/server.go [hunk 2]
```diff
@@ -51,6 +52,8 @@ type MinionServer struct {
 	exit           chan error    // to signal fatal errors
 
 	pathOverride string // the PATH environment for the sub-processes
+	cgroupPrefix string // e.g. /mesos
```
Comment (sttts) [inline contrib/mesos/pkg/minion/server.go]: the default is `mesos`, not `/mesos`. Which one is correct?

File: contrib/mesos/pkg/minion/server.go [hunk 3]
```diff
@@ -257,6 +283,13 @@ func (ms *MinionServer) Run(hks hyperkube.Interface, _ []string) error {
 	}
 	ms.clientConfig = clientConfig
 
+	// derive the executor cgroup and use it as:
+	// - pod container cgroup root (e.g. docker cgroup-parent)
+	// - parent of kubelet container
+	// - parent of kube-proxy container
+	ms.cgroupRoot = findMesosCgroup(ms.cgroupPrefix)
+	log.Infof("using cgroup-root %q", ms.cgroupRoot)
```
Comment (sttts) [inline contrib/mesos/pkg/minion/server.go]: Maybe reduce loglevel to 2 like for the findMesosCgroup logs

Comment (sttts) [inline contrib/mesos/pkg/minion/server.go]: And make a warning out of it if the cgroupRoot is empty?

File: contrib/mesos/pkg/scheduler/service/service.go [hunk 4]
```diff
@@ -162,10 +162,10 @@ func NewSchedulerServer() *SchedulerServer {
 
 		RunProxy:                 true,
 		ExecutorSuicideTimeout:   execcfg.DefaultSuicideTimeout,
-		ExecutorCgroupPrefix:     execcfg.DefaultCgroupPrefix,
 		DefaultContainerCPULimit: mresource.DefaultDefaultContainerCPULimit,
 		DefaultContainerMemLimit: mresource.DefaultDefaultContainerMemLimit,
 
+		ExecutorCgroupPrefix:  minioncfg.DefaultCgroupPrefix,
```
Comment (sttts) [inline contrib/mesos/pkg/scheduler/service/service.go]: -> MinionCgroupPrefix

File: contrib/mesos/pkg/minion/server.go [hunk 5]
```diff
@@ -132,6 +153,11 @@ func (ms *MinionServer) launchExecutorServer() {
 	ms.AddExecutorFlags(executorFlags)
 	executorArgs, _ := filterArgsByFlagSet(allArgs, executorFlags)
 
+	executorArgs = append(executorArgs, "--resource-container="+path.Join("/", ms.cgroupRoot, "kubelet"))
```
Comment (sttts) [inline contrib/mesos/pkg/minion/server.go]: Shouldn't this go into the if-clause as well?

File: contrib/mesos/pkg/minion/server.go (hunk 2 above)
Comment (jdef) [inline contrib/mesos/pkg/minion/server.go]: mesos

File: contrib/mesos/pkg/minion/server.go (hunk 3 above)
Comment (jdef) [inline contrib/mesos/pkg/minion/server.go]: dislike V(2) for this. :+1: for warning if empty

File: contrib/mesos/pkg/minion/server.go (hunk 5 above)
Comment (jdef) [inline contrib/mesos/pkg/minion/server.go]: no

Comment (jdef) [inline contrib/mesos/pkg/minion/server.go]: my intent was specifically to provide `--resource-container` all the time, and only specify `--cgroup-root` when there actually was one to specify.

**Comment by jdef** [comment]
needs squash

**Comment by sttts** [comment]
lgtm

**Comment by jdef** [comment]
@ixdy I think the k8s-merge-robot is spamming this thread

**Comment by jdef** [comment]
@davidopp I'm happy to add the 'ok-to-merge' label. Was under the impression that I should only be adding 'lgtm' and that oncall would merge. If that process has changed, please let me know.

**Comment by ixdy** [comment]
@brendandburns is the maintainer of the merge bot, and wow that was annoying.

**Pull request merged**